org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/jdt/JavaDebugServer.java

```diff
@@ -99,14 +97,20 @@ private void closeServerSocket() {
         serverSocket = null;
     }
 
-    private void closeConnection() {
-        if (connection != null) {
+    private void createConnectionThread(Socket connection) {
+        new Thread(() -> {
```

Comment:
The new thread is left unmanaged and unreferenced. I think this function should return the new thread. And the thread should be stopped/interrupted in stop().